filer/admin/folderadmin.py

```diff
@@ -1130,12 +1142,17 @@ def files_toggle_restriction(self, request, restriction,
         if request.method != 'POST':
             return None
         # cannot restrict/unrestrict unfiled files
-        if files_qs.filter(folder__isnull=True).exists():
-            raise PermissionDenied
+        unfiled_files = files_qs.filter(folder__isnull=True)
+        if bool(unfiled_files):
```

Comment:
I used bool so a second query will not be made if there are any files in this situations. but since this is a very improbable case I should change it.